### PR TITLE
LPS-163238 [Raylife AP] Creating Product Performance Product Filters

### PR DESCRIPTION
**Pull request by eduardo-diniz**
Task: [LPS-163238](https://issues.liferay.com/browse/LPS-163238)

### CODE REVIEW

**Comment by eliasantosf** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
Task: [LPS-163238](https://issues.liferay.com/browse/LPS-163238)

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts [hunk 1]
```diff
@@ -75,35 +112,64 @@ export async function annualRule(
...
 		currentDateString[0],
 		currentDateString[1],
 		currentDateString[0],
 		january
 	);
-	const yearToDateSalesResult = getArrayOfSales(
-		policiesSalesGoals,
-		yearToDateSalesArray
+
+	const productsType = salesGoal.data.items.filter((item: SalesGoal) => {
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts:132]:
```suggestion
	const productsType = salesGoal?.data?.items?.filter((item: SalesGoal) => {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts [hunk 2]
```diff
@@ -75,35 +112,64 @@ export async function annualRule(
...
+		if (
+			item.productExternalReferenceCode === productExternalReferenceCode
+		) {
+			return item;
+		}
+		else if (productExternalReferenceCode === 'All') {
+			return item;
+		}
+	});
+	const productsSalesType = policiesSalesGoals.data.items.filter(
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts:142]:
```suggestion
	const productsSalesType = policiesSalesGoals?.data?.items?.filter(
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts (hunk 1 above)
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts:132]: How are you using this more than once, maybe you should put in a variable like
```suggestion
	const getSalesGoalData = salesGoal?.data?.items
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts [hunk 3]
```diff
@@ -114,52 +180,107 @@ export async function sixMonthRule(
...
+				item.productExternalReferenceCode ===
+				productExternalReferenceCode
+			) {
+				return item;
+			}
+			else if (productExternalReferenceCode === 'All') {
+				return item;
+			}
+		}
+	);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts:272]: You should use a generic function for this like:
```suggestion
   function compareProducts (data) {
    data.filter((item: dataItem) => {
		if (
			item.productExternalReferenceCode === productExternalReferenceCode
		) {
			return item;
		}
		else if (productExternalReferenceCode === 'All') {
			return item;
		}
	});
	
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx [hunk 4]
```diff
@@ -453,6 +482,7 @@ const ProductPerformance = () => {
 							displayType="unstyled"
 							onClick={() => {
 								if (!products.every(isFilterAllActive)) {
+									setProductsValues('All');
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx:485]: It's a good practice to use a variable for this like:
```suggestion
					const inicialProductValue= 'All';
					setProductsValues(inicialProductValue);
```

**Review by joanagrego — CHANGES_REQUESTED** [review]
Good job! I added some suggestions :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx (hunk 4 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/index.tsx:485]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts (hunk 1 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts:132]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts (hunk 3 above)
Comment (eduardo-diniz) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/businessRules/index.ts:272]: Done. 
Thx!!

**Review by silasbsi — CHANGES_REQUESTED** [review]

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/ProductPerfomanceTypes.ts [hunk 5]
```diff
@@ -21,25 +21,45 @@ export type DataChart = {
...
-};
-
 export type Policy = {
 	boundDate: string;
 	productExternalReferenceCode: string;
 	productName: string;
 	termPremium: number;
 };
 
+export type Month = [
```
[the remark this answers is not in the document]
Comment (silasbsi) [reply modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/dashboard/ProductPerfomance/ProductPerfomanceTypes.ts]: it is missing propagate or adjust for all types.

**Pull request closed without merging**